src/supernova/mod.rs

```diff
@@ -938,9 +942,14 @@ where
   C1: StepCircuit<G1::Scalar>,
   C2: StepCircuit<G2::Scalar>,
 {
-  /// Initial program counter, defaults to zero.
+  /// Initial program counter is the initial circuit index as a `Scalar`.
   fn initial_program_counter(&self) -> G1::Scalar {
-    G1::Scalar::ZERO
+    G1::Scalar::from(self.initial_circuit_index() as u64)
+  }
+
+  /// Initial circuit index, defaults to zero.
+  fn initial_circuit_index(&self) -> usize {
+    0
```

Comment:
This API allows de-synchronizing the `usize` and `Scalar` representations of the initial program counter. It would be better to add `initial_program_counter` as an extension trait (a trait with a blanket implementation for `T: NonUniformCircuit`), then Rust will guarantee *that* implementation is the only one that can apply.

Reply:
Done.